src/systems/tunnel-effect.js

```diff
@@ -0,0 +1,140 @@
+import "../utils/postprocessing/EffectComposer";
+import "../utils/postprocessing/RenderPass";
+import "../utils/postprocessing/ShaderPass";
+import "../utils/postprocessing/MaskPass";
+import "../utils/shaders/CopyShader";
+import "../utils/shaders/VignetteShader";
+
+const STATIC = new THREE.Vector3(0, 0, 0);
+const CLAMP_VELOCITY = 0.01;
+
+AFRAME.registerSystem("tunneleffect", {
+  schema: {
+    targetComponent: { type: "string", default: "character-controller" },
+    movingEvent: { type: "string", default: "renderstart" },
+    radius: { type: "number", default: 0.45, min: 0.25 },
+    minRadius: { type: "number", default: 0.2, min: 0.1 },
+    softness: { type: "number", default: 0.1, min: 0.0 },
+    opacity: { type: "number", default: 1, min: 0.0 }
+  },
+
+  init: function() {
+    const data = this.data;
+    this.scene = this.el;
+    this.isMoving = false;
+    this.dt = 0;
+    this.t = 0;
+    this.radius = data.radius;
+    this.minRadius = data.minRadius;
+    this.softness = data.softness;
+    this.opacity = data.opacity;
+    this.characterVelocity = new THREE.Vector3(0, 0, 0);
+
+    // add event listener for init composer
+    this.characterEl = document.querySelector(`a-entity[${this.data.targetComponent}]`);
+    if (this.characterEl) {
+      this._initPostProcessing = this._initPostProcessing.bind(this);
+      this.characterEl.addEventListener("componentinitialized", this._initPostProcessing);
+    }
+  },
+
+  pause: function() {
+    if (!this.characterEl) {
+      return;
+    }
+    this.characterEl.removeEventListener("componentinitialized", this._initPostProcessing);
+  },
+
+  tick: function(time, deltaTime) {
+    this.t = time;
+    this.dt = deltaTime;
+
+    if (!this._isPostProcessingReady()) {
+      return;
+    }
+
+    this.characterVelocity = this.characterComponent.velocity;
+    if (this.characterVelocity.distanceTo(STATIC) < CLAMP_VELOCITY) {
+      // the character stops, so we use the aframe default render func
+      this.scene.renderer.render = this.originalRenderFunc;
+      this.isMoving = false;
```

Comment:
The effect seems to "pop" at the end, when you stop moving. Can we fade away quickly and continuously? It only seems to be a problem when I _stop_ moving. When I _start_ moving, it seems to fade in nicely.